src/pages/Admin/AdminUpdateClass/AdminUpdateClass.tsx

```diff
@@ -12,11 +12,42 @@ import {
 } from './AdminUpdateClass.styled';
 import ClassComponent from './ClassComponent';
 import { useModal } from '../../../components/Modal/useModal';
-import { Modal } from '../../../components/Modal/Modal';
-import { PrimaryButton } from '../../../components/Button';
+
+import { ErrorMessageType } from '../../../types';
+import { ERROR_MESSAGES } from '../../../utilities/constants';
+import { getAdminClass } from '../../../api/getAdminClasses';
+import { AddNewClassForm } from './AddNewClassModal';
+
+interface ClassComponentDataProps {
+  _id: string;
+  date: string;
+  name: string;
+  subject: string;
+  classworks?: [];
+}
 
 const AdminUpdateClass = () => {
   const { isOpen, toggle } = useModal();
+  const [classes, setClasses] = useState<ClassComponentDataProps[]>();
+  const { id } = useParams();
+
+  const fetchAdminClasses = useCallback(async () => {
+    try {
+      const res = await getAdminClass(id);
+      setClasses(res.data);
+    } catch (error) {
+      const errors = error as ErrorMessageType[];
+      if (errors?.[0]?.msg === ERROR_MESSAGES.unauthorized) {
+        // navigate('/sign-in', { replace: true });
```

Comment:
you can delete this and replace it with any other error handling :) I believe this part should be covered by the session cookie later on